src/client/notes.rs

```diff
@@ -15,6 +20,39 @@ impl<N: NodeRpcClient, R: FeltRng, S: Store> Client<N, R, S> {
         self.store.get_input_notes(filter).map_err(|err| err.into())
     }
 
+    /// Returns input notes that are able to be consumed by the account_id.
+    ///
+    /// If account_id is None then all consumable input notes are returned.
+    pub fn get_consumable_notes(
+        &self,
+        account_id: Option<AccountId>,
+    ) -> Result<Vec<ConsumableNote>, ClientError> {
+        let commited_notes = self.store.get_input_notes(NoteFilter::Committed)?;
+
+        let note_screener = NoteScreener::new(&self.store);
+        let mut relevant_notes: Vec<ConsumableNote> = Vec::new();
+
+        for input_note in commited_notes {
+            let account_relevance =
+                note_screener.check_relevance(&input_note.clone().try_into()?)?;
+            if account_relevance.is_empty() {
+                continue;
+            }
+            if let Some(account_id) = account_id {
+                if account_relevance.iter().any(|(id, _)| *id == account_id) {
+                    relevant_notes.push((
+                        input_note,
+                        account_relevance.into_iter().filter(|(id, _)| *id == account_id).collect(),
+                    ));
+                }
+            } else {
+                relevant_notes.push((input_note, account_relevance));
+            }
+        }
```

Comment:
What do you think about this? The idea is to split the filter by Id from the creation of the `ConsumableNote`s. It removes more nested Ifs, although I'm doubting whether the end result is more understandable or not.

```rust
        let mut relevant_notes = Vec::new();
        for input_note in commited_notes {
            let account_relevance =
                note_screener.check_relevance(&input_note.clone().try_into()?)?;

            if account_relevance.is_empty() { continue; }

            relevant_notes.push((input_note, account_relevance));
        }

        // If no account id was provided we already have the result
        if account_id.is_none() { return Ok(relevant_notes); }
        let account_id = account_id.expect("Should retrieve account id");

        // Use the given id to filter notes
        let relevant_notes = relevant_notes
            .into_iter()
            .filter(|(_input_note, relevance)| relevance.iter().any(|(id, _)| *id == account_id))
            .collect();

        Ok(relevant_notes)
```